src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -189,29 +189,56 @@ private function replaceByShortType(Tokens $tokens, TypeAnalysis $type, array $u
      */
     private function getTypes(Tokens $tokens, int $index, int $endIndex): iterable
     {
-        $index = $typeStartIndex = $typeEndIndex = $tokens->getNextMeaningfulToken($index - 1);
-        $type = $tokens[$index]->getContent();
-
+        $skipNextYield = false;
+        $typeStartIndex = $typeEndIndex = null;
+        $type = null;
         while (true) {
-            $index = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$index]->isGivenKind(CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN)) {
+                $index = $tokens->getNextMeaningfulToken($index);
+                $typeStartIndex = $typeEndIndex = null;
+                $type = null;
+
+                continue;
+            }
 
-            if ($tokens[$index]->isGivenKind([CT::T_TYPE_ALTERNATION, CT::T_TYPE_INTERSECTION])) {
-                yield $type => [$typeStartIndex, $typeEndIndex];
+            if (
+                $tokens[$index]->isGivenKind([CT::T_TYPE_ALTERNATION, CT::T_TYPE_INTERSECTION, CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE])
+                || $index > $endIndex
+            ) {
+                if (!$skipNextYield && null !== $typeStartIndex) {
+                    $origCount = \count($tokens);
+
+                    yield $type => [$typeStartIndex, $typeEndIndex];
+
+                    $endIndex += \count($tokens) - $origCount;
+
+                    // type tokens were possibly updated, restart type match
+                    $skipNextYield = true;
+                    $index = $typeEndIndex = $typeStartIndex;
+                    $type = null;
+                } else {
+                    $skipNextYield = false;
+                    $index = $tokens->getNextMeaningfulToken($index);
+                    $typeStartIndex = $typeEndIndex = null;
+                    $type = null;
+                }
 
-                $index = $typeStartIndex = $typeEndIndex = $tokens->getNextMeaningfulToken($index);
-                $type = $tokens[$index]->getContent();
+                if ($index > $endIndex) {
+                    break;
+                }
 
                 continue;
             }
 
-            if ($index > $endIndex || !$tokens[$index]->isGivenKind([T_STRING, T_NS_SEPARATOR])) {
-                yield $type => [$typeStartIndex, $typeEndIndex];
-
-                break;
+            if (null === $typeStartIndex) {
+                $typeStartIndex = $index;
+                $type = '';
```

Comment:
Is it required here? It's instantly overridden in line 239.

Reply:
Good catch! I was refactoring this code several times and reassigning the same var again is of course not needed... Fixed.

Reply:
Bad catch actually, in l239, we set end index instead of start index, had to revert. Never mind, I was blind too when landing it :D

Reply:
I was not talking about `$typeStartIndex` (yeah, I was aware it's different variable), but about `$type`.. But now I see there's `.=` not `=` so it's not overriding, but resetting before appending content. I was wrong 😉.